convert2rhel/backup.py

```diff
@@ -434,10 +437,17 @@ def remove_pkgs(
         loggerinst.info("Removing package: %s" % nvra)
         _, ret_code = run_subprocess(["rpm", "-e", "--nodeps", nvra])
         if ret_code != 0:
-            if critical:
-                loggerinst.critical("Error: Couldn't remove %s." % nvra)
-            else:
-                loggerinst.warning("Couldn't remove %s." % nvra)
+            pkgs_failed_to_remove.append(nevra)
+        else:
+            pkgs_removed.append(nevra)
+
+    if pkgs_failed_to_remove:
+        if critical:
+            loggerinst.critical("Error: Couldn't remove %s." % ", ".join(pkgs_failed_to_remove))
```

Comment:
In the action that calls this function - handle_packages, we have a try/except block that will catch the logger critical and set an error result.
try:
            pkgs_to_remove = sorted(pkghandler._get_packages_to_remove(system_info.excluded_pkgs))
            pkgs_removed = sorted(pkghandler.remove_pkgs_unless_from_redhat(pkgs_to_remove))

            # TODO: Handling SystemExit here as way to speedup exception
            # handling and not refactor contents of the underlying function.
        except SystemExit as e:
            # TODO(r0x0d): Places where we raise SystemExit and need to be
            # changed to something more specific.
            #   - When we can't remove a package.
            self.set_result(level="ERROR", id="EXCLUDED_PACKAGE_REMOVAL_FAILED", message=str(e))
            return

Reply:
@pr-watson I think Rodolfo is saying that we can make this code raise a more specific exception than `SystemExit`.  Then in handle_packages, we can add another `except NewException:` block which does the proper `set_result` with a more accurate id and message.

Reply:
After discussion with Toshio we have agreed to leave this function as is since the logger critical was implemented before the action framework. For the specific exception being raised, that will be taken care of in a separate PR